apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -79,6 +79,54 @@ export class TaskUC {
 		return response;
 	}
 
+	async finishTask(userId: EntityId, taskId: EntityId): Promise<TaskWithStatusVo> {
+		const user = await this.userRepo.findById(userId, true);
+		const task = await this.taskRepo.findById(taskId);
+
+		if (!this.authorizationService.hasTaskPermission(user, task, TaskParentPermission.read)) {
+			throw new UnauthorizedException();
+		}
+
+		task.finishForUser(user);
+		await this.taskRepo.save(task);
+
+		// add status
+		const status = this.authorizationService.hasOneOfTaskDashboardPermissions(
+			user,
+			TaskDashBoardPermission.teacherDashboard
+		)
+			? task.createTeacherStatusForUser(user)
```

Comment:
Do we need the status for this operation?

Reply:
Yes. We need the result in nuxt.